src/frontend/js/modules/sidebar.js

```diff
@@ -0,0 +1,142 @@
+  }
+
+  /**
+   * Called by ModuleDispatcher to initialize module from DOM
+   *
+   * @param {writingSettings} settings - module settings
+   * @param {HTMLElement} moduleEl - module element
+   */
+  init(settings, moduleEl) {
+    this.nodes.sections = Array.from(moduleEl.querySelectorAll('.' + Sidebar.CSS.section));
+    this.nodes.sections.forEach(section => {
+      const id = section.getAttribute('data-id');
+      const togglerEl = section.querySelector('.' + Sidebar.CSS.toggler);
+
+      if (!togglerEl) {
+        return;
+      }
+
+      togglerEl.addEventListener('click', e => this.handleSectionTogglerClick(id, section, togglerEl, e));
+
+      if (typeof this.collapsed[id] === 'undefined') {
+        this.collapsed[id] = false;
+      }
+      if (this.collapsed[id]) {
+        this.setSectionCollapsed(section, togglerEl, true, false);
+      }
+    });
+    this.nodes.sidebar = moduleEl.querySelector('.' + Sidebar.CSS.sidebar);
+    this.nodes.toggler = moduleEl.querySelector('.' + Sidebar.CSS.sidebarToggler);
+    this.nodes.toggler.addEventListener('click', () => this.toggleSidebar());
+    this.ready();
+  }
+
+  /**
+   * Toggles section expansion
+   *
+   * @param {number} sectionId - id of the section to toggle
+   * @param {HTMLElement} sectionEl - section html element
+   * @param {HTMLElement} togglerEl - toggler button html element
+   * @param {MouseEvent} event - click event
+   * @returns {void}
+   */
+  handleSectionTogglerClick(sectionId, sectionEl, togglerEl, event) {
+    event.preventDefault();
+    this.collapsed[sectionId] = !this.collapsed[sectionId];
+    window.localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(this.collapsed));
+    this.setSectionCollapsed(sectionEl, togglerEl, this.collapsed[sectionId]);
+  }
+
+  /**
+   * Updates section's collapsed state
+   *
+   * @param {HTMLElement} sectionEl - element of the section to toggle
+   * @param {HTMLElement} togglerEl - section's toggler button element
+   * @param {boolean} collapsed - new collapsed state
+   * @param {boolean} animated - true if state should change with animation
+   */
+  setSectionCollapsed(sectionEl, togglerEl, collapsed, animated = true) {
+    togglerEl.classList.toggle(Sidebar.CSS.togglerCollapsed, collapsed);
+    const sectionList = sectionEl.querySelector('.' + Sidebar.CSS.sectionList);
+
+    if (!sectionList) {
+      return;
+    }
+    sectionEl.classList.toggle(Sidebar.CSS.sectionAnimated, animated);
+    sectionList.classList.toggle(Sidebar.CSS.sectionListCollapsed, collapsed);
+
+    /**
+     * Highlight section item as active if active child item is collapsed.
+     */
+    const activeSectionListItem = sectionList.querySelector('.' + Sidebar.CSS.sectionListItemActive);
+    const sectionTitle = sectionEl.querySelector('.' + Sidebar.CSS.sectionTitle);
+
+    if (!activeSectionListItem) {
+      return;
+    }
+    if (collapsed && animated) {
+      setTimeout(() => {
```

Comment:
pls, leave a comment. Could we use the requestAnimationFrame here?

Reply:
added a comment